data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -114,6 +139,40 @@ public void shutdown() {
 
     private void registerPatterns() {
         grokCompiler.registerDefaultPatterns();
+        grokCompiler.register(grokPrepperConfig.getPatternDefinitions());
+        registerPatternsDir();
+    }
+
+    private void registerPatternsDir() {
+        final Pattern compiledFilesGlob = Pattern.compile(grokPrepperConfig.getPatternsFilesGlob());
```

Comment:
A glob pattern is different from regex. A major difference is that `**` lets you match going into directories.

I believe this is what you need:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newDirectoryStream-java.nio.file.Path-java.lang.String-